templates/init.html

```diff
@@ -91,6 +91,9 @@
 }
 
   </style>
+  <script src="/static/codemirror/lib/codemirror.js"></script>
```

Comment:
You have mentioned `static/codemirror` folder here. But it hasn't been added in your PR.

Reply:
![image](https://user-images.githubusercontent.com/11318551/34210192-ffc3e612-e5ba-11e7-96a0-66b5036a8efe.png)

Reply:
But it is there locally  :confused:

Reply:
You have mentioned `codemirror` in .gitignore. Seems it's ignoring all codemirror folders.

I think you wanted to ignore `./codemirror`. but gitignore makes it ignore `./static/codemirror` as well. Check how you can fix this. Looks like an easy fix.

Reply:
Is this because i have added codemiror in .gitignore?

Reply:
if i remove codemirror from .gitignore there will be  a lot of files then.

Reply:
You need to see how to add `./codemirror` to .gitignore, but not ignore `./static/codemirror`

It is possible to NOT ignore certain files / folders in .gitignore. What you've done is not wrong, it's just incomplete.

Reply:
./codemirror
!./static/codemirror
will do the work i guess